### PR TITLE
Add boltdb cache

### PR DESCRIPTION
**Pull request by bored-engineer**

### CODE REVIEW

**Comment by bored-engineer** [comment]
Fixed build issues.

**Comment by bored-engineer** [comment]
Might be worth some discussion of `Batch` vs `Update`

**Comment by shurcooL** [comment]
Looks reasonable to me.

It's up to @gregjones whether he wants to accept more implementations inside this repo, or ask you to maintain it out of repo (and add a link to README).

File: boltdbcache/boltdbcache_test.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+func TestBoltDBCache(t *testing.T) {
+	tempDir, err := ioutil.TempDir("", "httpcache")
+	if err != nil {
+		t.Fatalf("TempDir: %v", err)
+	}
+	defer os.RemoveAll(tempDir)
+
+	cache, err := New(filepath.Join(tempDir, "db"), []byte("cache"))
+	if err != nil {
+		t.Fatalf("New leveldb,: %v", err)
```
Comment (gregjones) [inline boltdbcache/boltdbcache_test.go]: s/leveldb/boltdb

File: boltdbcache/boltdbcache.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+// Delete removes the response with key from the cache
+func (c *Cache) Delete(key string) {
+	c.db.Batch(func(tx *bolt.Tx) error {
+		bucket := tx.Bucket(c.bucket)
+		return bucket.Delete([]byte(key))
+	})
+}
+
+// New returns a new Cache that will store a boltdb in path using bucket (defaults to "cache")
```
Comment (gregjones) [inline boltdbcache/boltdbcache.go]: The comment doesn't seem accurate - not providing a bucket results in an error.

**Comment by gregjones** [comment]
I'm happy with it living here.
I've added a couple of minor comments.
One question though: I've never used boltdb, but the DB type has a Close method. Should we allow users to call that?

**Comment by shurcooL** [comment]
@bored-engineer Did you see the question Greg asked above, and minor comments?

This PR seems close to being merge-able. Just following up here since it's been a while.

**Comment by shurcooL** [comment]
Ping @bored-engineer. Did you see my comment above?

**Comment by gregjones** [comment]
If you want to come back to this at some point, please feel free to reopen it!

**Comment by birkelund** [comment]
I made a [coreos/bbolt](https://github.com/coreos/bbolt) based backend at [birkelund/boltdbcache](https://github.com/birkelund/boltdbcache).

Feel free to copy and paste it if you would like it in-tree.

**Pull request closed without merging**